src/MSBuild/XMake.cs

```diff
@@ -1233,7 +1233,7 @@ string outputResultsCache
 
                             if (!restoreOnly)
                             {
-                                if (graphBuild)
+                                if (graphBuildOptions != null)
```

Comment:
`&& graphBuildOptions.Build`?

Reply:
`graphBuildOptions.Build` gets acted upon later on in the BuildManager, after the graph is constructed. That's the whole point, construct the graph but not build it.